### PR TITLE
Add optional handling of Nyx InvalidWriteToPayload event

### PR DESCRIPTION
**Pull request by choller**
This adds `AFL_NYX_HANDLE_INVALID_WRITE` to allow AFL++ to ignore the InvalidWriteToPayload event. If such events occur sporadically, it is most likely due to OOM in the guest where the kernel is reclaiming memory and therefore touching the payload buffer. I decided to count these as crashes, hoping that they remain visible with this approach (but with a zero-size log). However, this happens quite rarely in my local setup so it is difficult to test.

Without this variable set, AFL++ will just die and leave qemu running in the background. Long term, we might want to change the default behavior here as well.

### CODE REVIEW

**Comment by vanhauser-thc** [comment]
> Without this variable set, AFL++ will just die and leave qemu running in the background. Long term, we might want to change the default behavior here as well.

but it is wrong to keep it like this. this should kill the target (qemu). can you please add this?

File: src/afl-forkserver.c [hunk 1]
```diff
@@ -1840,6 +1840,8 @@ afl_fsrv_run_target(afl_forkserver_t *fsrv, u32 timeout,
       case Timeout:
         return FSRV_RUN_TMOUT;
       case InvalidWriteToPayload:
+        if (!!getenv("AFL_NYX_HANDLE_INVALID_WRITE")) { return FSRV_RUN_CRASH; }
+
         /* ??? */
         FATAL("FixMe: Nyx InvalidWriteToPayload handler is missing");
```
Comment (vanhauser-thc) [inline src/afl-forkserver.c:1846]: If you switch ` FATAL`  to `NYX_PRE_FATAL` it should be enough.

Comment (choller) [reply src/afl-forkserver.c:1846]: Thanks, I'm testing this right now locally.

**Comment by choller** [comment]
Fwiw, I just managed to reproduce with the new code (and the env var set). The log will contain something like:

`Payload buffer write attempt at RIP: ffffffff8dc26847`

So that's even nicer.

File: src/afl-forkserver.c (hunk 1 above)
Comment (vanhauser-thc) [reply src/afl-forkserver.c:1846]: great, ping when you have a result :)

**Comment by vanhauser-thc** [comment]
@choller @schumilo  any idea how to solve this gracefully?

**Comment by vanhauser-thc** [comment]
@choller @schumilo please continue to look for a fix ... merging now.

**Pull request merged**